src/WebJobs.Script/Description/Node/NodeFunctionInvoker.cs

```diff
@@ -532,6 +551,82 @@ internal static bool IsEdgeSupportedType(Type type)
             return requestObject;
         }
 
+        private static Dictionary<string, object> CreateResponseObject(HttpResponseMessage response, out string rawBody)
+        {
+            rawBody = null;
+
+            Dictionary<string, object> responseObject = new Dictionary<string, object>();
+
+            responseObject["statusCode"] = response.StatusCode;
+            responseObject["reasonPhrase"] = response.ReasonPhrase.ToString();
+
+            // since HTTP headers are case insensitive, we lower-case the keys
+            var headers = GetRawResponseHeaders(response).ToDictionary(p => p.Key.ToLowerInvariant(), p => p.Value);
+            responseObject["headers"] = headers;
+
+            if (response.Content != null && response.Content.Headers.ContentLength > 0)
+            {
+                MediaTypeHeaderValue contentType = response.Content.Headers.ContentType;
+                object jsonObject;
+                object body = null;
+                if (contentType != null)
+                {
+                    if (contentType.MediaType == "application/json")
+                    {
+                        body = response.Content.ReadAsStringAsync().Result;
+                        if (TryConvertJson((string)body, out jsonObject))
+                        {
+                            // if the content - type of the response is json, deserialize into an object or array
+                            rawBody = (string)body;
+                            body = jsonObject;
+                        }
+                    }
+                    else if (contentType.MediaType == "application/octet-stream")
+                    {
+                        body = response.Content.ReadAsByteArrayAsync().Result;
+                    }
+                }
+
+                if (body == null)
+                {
+                    // if we don't have a content type, default to reading as string
+                    body = rawBody = response.Content.ReadAsStringAsync().Result;
+                }
+
+                responseObject["responseBody"] = body;
+
+                var contentHeaders = response.Content.Headers.ToDictionary(p => p.Key.ToLowerInvariant(), p => p.Value.FirstOrDefault());
+                responseObject["contentHeaders"] = contentHeaders;
+            }
+
+            if(response.RequestMessage != null)
+            {
+                string rawRequestBody = null;
+                var requestObject = CreateRequestObject(response.RequestMessage, out rawRequestBody);
+                responseObject["request"] = requestObject;
+            }
+
+            return responseObject;
+        }
+
+        private static IDictionary<string, string> GetRawResponseHeaders(HttpResponseMessage response)
```

Comment:
This looks like a copy of the code in the HttpRequestMessage.GetRawHeaders extension method. We'd probably just add a parallel HttpResponseMessage extension method, and share the common code between the two.